### PR TITLE
feat: implement zipkin tracing

### LINKED ISSUE #1827: Enable selecting zipkin tracing provider
*Description*:
Currently Envoy Gateway only supports the OpenTelemetry tracer, this makes EG unsuitable to plug into platforms where other solutions are being used (especially regarding trace context propagation). Some may want to transition to OpenTelemetry in the future but others have significant investments in other solutions (often using the B3 context propagation standard from the Zipkin ecosystem) or use tracing ecosystems that simply can't be supported by OpenTelemetry (e.g. Apache SkyWalking).

As an ingress typically is the location where also the sampling decision is made, it is very important to support at least the core instrumentation that Envoy supports. At the very least start with Zipkin (for B3 propagation) and SkyWalking (native instrumentation is not OTel compatible)

### PR DESCRIPTION
**Pull request by alexandermarston**
**What type of PR is this?**
feat: implement zipkin tracing

**What this PR does / why we need it**:
This PR adds support for the Zipkin tracer.

**Which issue(s) this PR fixes**:
Fixes #1827 

API Change: #3630

### CODE REVIEW

File: internal/xds/translator/tracing.go [hunk 1]
```diff
@@ -23,26 +24,61 @@ import (
...
+			return protocov.ToAnyWithError(config)
+		}
+	case egv1a1.TracingProviderTypeZipkin:
+		providerName = envoyZipkin
+
+		providerConfig = func() (*anypb.Any, error) {
+			config := &tracecfg.ZipkinConfig{
+				CollectorCluster:         tracing.Destination.Name,
+				CollectorEndpoint:        "/api/v2/spans",
+				TraceId_128Bit:           *tracing.Provider.Zipkin.Enable128BitTraceID,
```
Comment (arkodg) [inline internal/xds/translator/tracing.go:68]: will crash if ptr is null

Comment (arkodg) [reply internal/xds/translator/tracing.go:68]: you'll need to use something like `ptr.Deref(<>, false)` as well as on L69

Comment (alexandermarston) [reply internal/xds/translator/tracing.go]: oversight on my part, should be handled now

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]
LGTM, thanks for adding this,
Would be better to add e2e test in the following.

File: internal/xds/translator/testdata/out/xds-ir/tracing-zipkin.clusters.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        region: tracing-0/backend/0
+  name: tracing-0
+  outlierDetection: {}
+  perConnectionBufferLimitBytes: 32768
+  respectDnsTtl: true
+  type: STRICT_DNS
+  typedExtensionProtocolOptions:
+    envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
+      '@type': type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
+      explicitHttpConfig:
```
Comment (zirain) [inline internal/xds/translator/testdata/out/xds-ir/tracing-zipkin.clusters.yaml:48]: this's not right, zipkin is HTTP.

**Pull request merged**